components/gantry/oneaxis/oneaxis.go

```diff
@@ -81,10 +82,10 @@ func init() {
 
 type oneAxis struct {
 	resource.Named
-	resource.AlwaysRebuild
 
 	board board.Board
 	motor motor.Motor
+	mu    sync.Mutex
```

Comment:
The mutex is only locked during reconfiguration, which is going to be useless because the coroutine that checks for config changes and calls `Reconfigure` is already a non-parallel. I suspect you need to also lock the mutex during all the other public functions that might interact with the state (`MoveToPosition`, etc.).

Reply:
So I need to add it to all public functions? Will do